configure.ac

```
@@ -6572,6 +6572,33 @@ then
   [Define if you have struct stat.st_mtimensec])
 fi
 
+# Check if year with century should be normalized for strftime
+AC_CACHE_CHECK([whether year with century should be normalized for strftime], [ac_cv_normalize_century], [
+AC_RUN_IFELSE([AC_LANG_SOURCE([[
+#include <stdlib.h>
+#include <time.h>
+#include <string.h>
+
+int main(void)
+{
+  char year[5];
+  struct tm date = {
+    .tm_year = -1801,
+    .tm_mon = 0,
+    .tm_mday = 1
+  };
+  exit(strftime(year, sizeof year, "%Y", &date) && !strcmp(year, "0099"));
```

You don't need `exit()` here; you can just use `return` and ditch the `stdlib.h` include. It is good practice to keep these C checks bare to the bone.

Moreover, I expect parentheses around the `sizeof` argument:

```suggestion
  if strftime(year, sizeof(year), "%Y", &date) && !strcmp(year, "0099")); {
    return 1;
  }
  return 0;
```

---

Right. Was mimicking the other existing C checks that call `exit` but I certainly agree that the main program can simply return instead. Also agree with the style suggestion of using parentheses after `sizeof`. Your suggested edit has a typo of a missing parenthesis and an extra semicolon though so I'm going to make my own commit for this. Thanks!